src/structure_managers/structure_manager.hh

```diff
@@ -580,7 +627,87 @@ namespace rascal {
       }
     }
 
-   protected:
+    template <size_t Order, size_t Layer,
+              bool HasDistances = traits::HasDistances,
+              typename std::enable_if_t<HasDistances, int> = 0>
+    inline double & get_distance(const ClusterRefKey<Order, Layer> & pair) {
+      return this->get_previous_manager()->get_distance(pair);
+    }
+
+    template <size_t Order, size_t Layer,
+              bool HasDistances = traits::HasDistances,
+              typename std::enable_if_t<not(HasDistances), int> = 0>
+    inline double & get_distance(const ClusterRefKey<Order, Layer> &) const {
+      throw std::runtime_error("Trying to get_distance from a manager stack "
+                               "without computed distances.");
+    }
+
+    template <size_t Order, size_t Layer,
+              bool HasDirectionVectors = traits::HasDirectionVectors,
+              typename std::enable_if_t<HasDirectionVectors, int> = 0>
+    inline const Vector_ref
+    get_direction_vector(const ClusterRefKey<Order, Layer> & pair) {
+      return this->get_previous_manager()->get_direction_vector(pair);
+    }
+
+    template <size_t Order, size_t Layer,
+              bool HasDirectionVectors = traits::HasDirectionVectors,
+              typename std::enable_if_t<not(HasDirectionVectors), int> = 0>
+    inline const Vector_ref
+    get_direction_vector(const ClusterRefKey<Order, Layer> &) const {
+      throw std::runtime_error("Trying to get direction vectors from a manager "
```

Comment:
Is it better to throw a runtime error or a compilation error ? I would say compilation by not implementing this function even if the compilation error is not so straightforward it is still understandable and avoids long compilations

Reply:
changed to static_assert, so the compiler error is more understandable

Reply:
had to remove these functions, because they are always compiled if HasDistance is false, even when they are not used anywhere